### ISSUE TITLE
`pnpm catalog:` protocol dependencies are not recognized or checked for updates

### ISSUE THREAD

**Issue by onemen** [raineorshine/npm-check-updates#1863]
### Description
`ncu` fails to detect outdated dependencies when they are defined using `pnpm` workspace catalogs (`catalog:` protocol). It erroneously reports that all dependencies match the latest versions, whereas `pnpm outdated` successfully identifies the available updates.

### Reproduction Files

**`package.json`**
```json
{
  "name": "test_ncu_with_catalog",
  "version": "1.0.0",
  "description": "test ncu with pnpm catalog",
  "main": "index.js",
  "type": "module",
  "license": "MIT",
  "devDependencies": {
    "eslint": "catalog:"
  }
}

```

**`pnpm-workspace.yaml`**

```yaml
catalog:
  eslint: 10.0.0

```

### Steps to Reproduce

1. Create a workspace with the files above.
2. Run `ncu --cooldown 1 --format cooldown,time,group`
3. Run `pnpm outdated`

#### 1. `ncu` Output (Incorrect Behavior with `catalog:`)

When using `"eslint": "catalog:"`:

```text
Using pnpm
Checking C:\code\test_ncu\package.json
[============================] 1/1 100%
All dependencies match the latest package versions :)

```

* **Explanation:** `ncu` parses `"eslint": "catalog:"` but does not resolve the version from `pnpm-workspace.yaml`'s catalog definitions. Because it doesn't recognize the `catalog:` specifier as a trackable local version string, it incorrectly concludes that nothing needs updating. It fails to report that `eslint 10.5.0` was published 1 week ago and should be flagged as an update.

#### 2. `ncu` Output (Correct Behavior with Explicit Version)

If the catalog protocol is bypassed and the version is defined explicitly in `package.json` like this:

```json
"devDependencies": {
  "eslint": "10.0.0"
}

```

`ncu` works perfectly and produces the expected output:

```text
Using pnpm
Checking C:\code\test_ncu\package.json
[====================] 1/1 100%

Minor   Backwards-compatible features
 eslint  10.0.0  →  10.5.0    1 week ago

```

#### 3. `pnpm outdated` Output (Expected Behavior with `catalog:`)

```text
┌──────────────┬─────────┬────────┐
│ Package      │ Current │ Latest │
├──────────────┼─────────┼────────┤
│ eslint (dev) │ 10.0.0  │ 10.5.0 │
└──────────────┴─────────┴────────┘

```

* **Explanation:** `pnpm` natively reads the catalog definition from `pnpm-workspace.yaml`, maps `catalog:` to version `10.0.0`, and successfully detects that a newer version (`10.5.0`) is available upstream on the registry.

### Expected Behavior

`ncu` should recognize the `catalog:` protocol, cross-reference it with the definitions in `pnpm-workspace.yaml`, and report that `eslint` can be updated (or update the version inside `pnpm-workspace.yaml` directly if running with `-u`).

### Environment

* **pnpm Version:** 11.8.0

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

- **Plain `ncu`** (single package) → never reads `pnpm-workspace.yaml`, so it can't resolve `catalog:` and reports "All dependencies match."
- **`ncu --workspaces`** → reads `pnpm-workspace.yaml`, but errors out because there's no `packages:` key.

A `pnpm-workspace.yaml` is perfectly valid with **only** a `catalog:` and no `packages:` — pnpm itself supports this (it's how single-package repos use catalogs). But `ncu` currently assumes `pnpm-workspace.yaml` implies an actual multi-package workspace.

## Workaround

Add an empty/self `packages:` entry to `pnpm-workspace.yaml` so `ncu --workspaces` proceeds:

````yaml name=pnpm-workspace.yaml
packages:
  - '.'

catalog:
  eslint: 10.0.0
````

Then `ncu --workspaces -u` should detect and bump the `eslint` catalog entry. (You may also want `--no-root` or `--root` depending on whether you want the root `package.json` checked too.)

## Suggested Fix

The proper fix is for `ncu` to recognize a `pnpm-workspace.yaml` that defines catalogs even when no `packages:` key exists — i.e., the catalog package info should be collected (and registry-checked) independently of the workspace-glob requirement, rather than short-circuiting with `programError` when `packages` is missing.

**Comment by onemen** [comment]
`ncu --format --cooldown,time --workspaces`

1. Why ncu read my .npmrc when it run in folder with `pnpm-workspace.yaml` and `pnpm-lock.yaml`
2. the output is doubled 
3. skip by cooldown is missing

```powershell
ncu --format --cooldown,time --workspaces
Using min-release-age from .npmrc: 5 days
Using pnpm
Checking C:\code\test_ncu\package.json
[====================] 2/2 100%
 eslint  10.0.0  →  10.5.0  [cooldown] 10.6.0    2 weeks ago
 vite     8.0.0  →  8.0.16  [cooldown] 8.1.0     3 weeks ago

Run ncu --format "--cooldown time" --workspaces -u to upgrade package.json

Checking C:\code\test_ncu\package.json
[====================] 2/2 100%
 eslint  10.0.0  →  10.5.0  [cooldown] 10.6.0    2 weeks ago
 vite     8.0.0  →  8.0.16  [cooldown] 8.1.0     3 weeks ago

Run ncu --format "--cooldown time" --workspaces -u to upgrade package.json

Checking C:\code\test_ncu\pnpm-workspace.yaml catalog dependencies
[====================] 1/1 100%
 eslint  10.0.0  →  10.5.0  [cooldown] 10.6.0    2 weeks ago

Run ncu --format "--cooldown time" --workspaces -u to upgrade pnpm-workspace.yaml

```

**Comment by raineorshine** [comment]
> 1. Why ncu read my .npmrc when it run in folder with pnpm-workspace.yaml and pnpm-lock.yaml

Thanks for pointing this out. This was an oversight from #1631. Opened #1882 to fix.

> 2. the output is doubled

I guess the `packages:  - '.'` workaround would require `ncu --workspaces --no-root`.

> 3. skip by cooldown is missing

Please open a separate issue for this.

**Comment by onemen** [comment]
Everything work as expected with ncu v 22.2.8
